docs/snippets/react/storybook-main-prop-filter.ts.mdx

```diff
@@ -0,0 +1,23 @@
+```ts
+// .storybook/main.ts
+
+// Replace your-framework with the framework you are using (e.g., react-webpack5, react-vite)
+import type { StorybookConfig } from '@storybook/your-framework';
+
+const config: StorybookConfig = {
+  framework: '@storybook/your-framework',
+  stories: ['../src/**/*.mdx', '../src/**/*.stories.@(js|jsx|mjs|ts|tsx)'],
+  typescript: {
+    reactDocgen: 'react-docgen-typescript',
+    reactDocgenTypescriptOptions: {
+      compilerOptions: {
+        allowSyntheticDefaultImports: false,
+        esModuleInterop: false,
+      },
+      propFilter: () => true,
```

Comment:
For most React projects, this will include all the props in `@types/react`, which includes _every_ valid HTML attribute (100s, for many elements).

Much better to scope it to the third-party package(s) they need, as we do in the [MUI recipe](https://storybook.js.org/recipes/@mui/material/#4-use-material-ui-prop-types-for-better-controls-and-docs):

```js
// Filter out third-party props from node_modules except @mui packages
propFilter: (prop) =>
  prop.parent ? !/node_modules\/(?!@mui)/.test(prop.parent.fileName) : true,
```

This has a nice benefit of communicating that `prop` is available as a function parameter.